go-controller/pkg/ovn/egressip.go

```diff
@@ -513,23 +514,33 @@ func (oc *Controller) reconcileCloudPrivateIPConfig(old, new *ocpcloudnetworkapi
 	}
 
 	if shouldDelete {
-		// There can only be one EgressIP object holding a given egress IP. If
-		// multiple objects specify the same egress IP the assignment algorithm
-		// invalidates it, hence: the following is safe.
-		egressIP, err := oc.getEgressIPFromCloudPrivateIPConfigName(oldCloudPrivateIPConfig.Name)
-		if err != nil {
+		// Get the EgressIP owner reference
+		egressIPName, exists := oldCloudPrivateIPConfig.Annotations[util.OVNEgressIPOwnerRefLabel]
+		if !exists {
+			return fmt.Errorf("CloudPrivateIPConfig object: %s is missing the egress IP owner reference annotation", oldCloudPrivateIPConfig.Name)
+		}
+		// Check if the egress IP has been deleted or not, if we are processing
+		// a CloudPrivateIPConfig delete because the EgressIP has been deleted
+		// then we need to remove the setup made for it, but not update the
+		// object.
+		egressIP, err := oc.kube.GetEgressIP(egressIPName)
```

Comment:
use informer instead of kubeclient

Reply:
See my above comments on the informer cache and multiple updates in a row for objects which we read and write from/to. I would rather use a extra network call than spend days debugging a difficult cache problem.